### DISCUSSION TITLE
$scroll not working with $watch

### DISCUSSION THREAD

**Discussion by blakeyman** [alpine-collective/alpine-magic-helpers#137] [Q&A]
Although not achieving behaviour expected the following is loading without error
`<div x-data="chat()" x-init="() => { listen(); $watch('messages', () => $scroll($refs.anchor) ); }">`

However, when I try and add additional parameters such as
`<div x-data="chat()" x-init="() => { listen(); $watch('messages', () => $scroll($refs.anchor, {behavior: auto}) ); }">`

Then I'm getting the following error
`Uncaught (in promise) ReferenceError: auto is not defined`

Surely this is a bug???

**Comment by KevinBatdorf** [comment]
You have `auto` the variable when you want the string `'auto'`

**Reply by blakeyman** [reply]
Thank you (although when my 'messages' load it doesn't scroll to anchor)
Is it different as it's in x-init ??? 
[edit: see post below]

**Reply by KevinBatdorf** [reply]
Where's the post below? 

is `$refs.anchor` visible on the page when you attempt to scroll?

**Reply by blakeyman** [reply]
no - it's within a `<div class="overflow-auto h-64">` 

I think I'm expecting it to scroll that `<div>` (which it's not)
But it is scrolling to the `<div>` - just not scrolling "that" actual one...
(I hope that makes sense?)

**Reply by blakeyman** [reply]
so the `$scroll` is working - just not my application!

**Reply by KevinBatdorf** [reply]
If you have a demo I can check it out to see what's going on

**Reply by blakeyman** [reply]
Tricky as it's local (and running own websocket server, laravel queues etc)
However, here is the "blade" file.
https://gist.github.com/blakeyman/71d0597cbbc08d83bfeb4e99f766600f

**Reply by KevinBatdorf** [reply]
Yeah It's not really easy to debug without seeing it. The code looks fine at a glance. Just try to narrow it down

**Comment by SimoTod** [comment]
Nope it's not a bug, it's a typo in the documentation. auto should be a string. (there isn't any variable called auto).
